app/models/queue_item.rb

```diff
@@ -0,0 +1,24 @@
+class QueueItem < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :video
+
+  validates_presence_of :user
+  validates_uniqueness_of :position, scope: :user_id
```

Comment:
I would not recommend this validation, when user updates queue items positions he will write the same positions and they won't be valid, so you ll have to mutate either inputs or existing positions. Better approach is when you save user positions as they are and then order and re-index. Instructor will show this in next lesson.